File: dev-personal-app/src/main/java/com/jesusterino/app/service/ReminderService.java
package com.jesusterino.app.service;

import com.jesusterino.app.dto.ReminderDTO;
import com.jesusterino.app.model.Reminder;
import com.jesusterino.app.model.User;
import com.jesusterino.app.repository.ReminderRepository;
import com.jesusterino.app.repository.UserRepository;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;

import java.util.List;
import java.util.stream.Collectors;

@Service
@RequiredArgsConstructor
public class ReminderService {

    private final ReminderRepository reminderRepository;
    private final UserRepository userRepository;

    public List<ReminderDTO> getReminders(String username) {
        User user = userRepository.findByUsername(username)
                .orElseThrow(() -> new RuntimeException("Usuario no encontrado"));
        return reminderRepository.findByUser(user).stream()
                .map(this::convertToDTO)
                .collect(Collectors.toList());
    }

    public ReminderDTO addReminder(String username, ReminderDTO dto) {
        User user = userRepository.findByUsername(username)
                .orElseThrow(() -> new RuntimeException("Usuario no encontrado"));

        Reminder reminder = Reminder.builder()
                .text(dto.getText())
                .color(dto.getColor())
                .startTime(dto.getStartTime())
                .endTime(dto.getEndTime())
                .allDay(dto.isAllDay())
                .user(user)
                .build();

        return convertToDTO(reminderRepository.save(reminder));
    }

    public void deleteReminder(Long id) {
        reminderRepository.deleteById(id);
    }

    private ReminderDTO convertToDTO(Reminder reminder) {
        return ReminderDTO.builder()
                .id(reminder.getId())
                .text(reminder.getText())
                .color(reminder.getColor())
                .startTime(reminder.getStartTime())
                .endTime(reminder.getEndTime())
                .allDay(reminder.isAllDay())
                .build();
    }
}

File: dev-personal-frontend/src/services/reminder.service.ts
import axios from 'axios';

const API_URL = 'http://localhost:8080/api/reminders/';

export interface Reminder {
  id: number;
  text: string;
  color: string;
  startTime: string; // ISO LocalDateTime
  endTime: string;   // ISO LocalDateTime
  allDay: boolean;
}

const getReminders = async (username: string): Promise<Reminder[]> => {
  const response = await axios.get(API_URL + username, { withCredentials: true });
  return response.data;
};

const addReminder = async (username: string, reminder: Partial<Reminder>): Promise<Reminder> => {
  const response = await axios.post(API_URL + username, reminder, { withCredentials: true });
  return response.data;
};

const deleteReminder = async (id: number): Promise<void> => {
  await axios.delete(API_URL + id, { withCredentials: true });
};

const ReminderService = {
  getReminders,
  addReminder,
  deleteReminder,
};

export default ReminderService;
